video/out/wayland_common.c

```diff
@@ -1134,8 +1135,16 @@ int vo_wayland_init(struct vo *vo)
         .cursor_visible = true,
         .vo_opts_cache = m_config_cache_alloc(wl, vo->global, &vo_sub_opts),
     };
+    wl->opts = mp_get_config_group(wl, wl->vo->global, &wayland_conf);
     wl->vo_opts = wl->vo_opts_cache->opts;
 
+    wl->display = wl_display_connect(wl->opts->display_socket);
+    if (!wl->display) {
+        MP_WARN(wl, "Display socket %s not found/unavailable! Falling back to NULL!\n", wl->opts->display_socket);
+        wl->display = wl_display_connect(NULL);
+    }
+    wl->display_fd = wl_display_get_fd(wl->display);
```

Comment:
you should probably do this after `if (!wl->display) return false`

Reply:
True